### PR TITLE
Update HTTP documentation with WebSockets

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 1]
```diff
@@ -36,6 +36,49 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
 
 See [Caching Example](https://ballerinalang.org/docs/by-example/caching), [HTTP Disable Chunking Example](https://ballerinalang.org/docs/by-example/http-disable-chunking).
 
+###WebSockets
+The package also provides WebSocket support. WebSocket endpoints have two types as `WebSocketClient` and `WebSocketListener`. Both types support ping/pong control frames and `WebSocketClient` supports callback services.
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: WebSocket support or support for WebSocket?

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 2]
```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
 
 See [Caching Example](https://ballerinalang.org/docs/by-example/caching), [HTTP Disable Chunking Example](https://ballerinalang.org/docs/by-example/http-disable-chunking).
 
+###WebSockets
+The package also provides support for WebSockets. There are two types of WebSocket endpoint which are `WebSocketClient` and `WebSocketListener`. Both endpoints support all WebSocket frames. The `WebSocketClient` has a callback service.
```
Comment (appellj) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Change "There are two types of WebSocket endpoint which are `WebSocketClient` and `WebSocketListener`." to "There are two types of WebSocket endpoints: `WebSocketClient` and `WebSocketListener`."

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 3]
```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
...
+
+There are also two types of services in WebSocket which are `WebSocketService` and `WebSocketClientService`. The callback service for `WebSocketClient` would always be a `WebSocketClientService`. The WebSocket services have a fixed set of resources which do not have a resource config. The incoming messages are passed to these resources.
```
Comment (appellj) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Same construction here: "There are also two types of services in WebSocket: `WebSocketService` and `WebSocketClientService`."

Comment (appellj) [reply stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Change "would always be" to "is always"

Comment (appellj) [reply stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Change "resources which" to "resources that"

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 4]
```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
...
+
+**WebSocket upgrade**: During a WebSocket upgrade the initial message would be an HTTP request. If this request is to be intercepted and the handshake be made explicitly by the user with custom headers the user has to create a HTTP resource with WebSocket specific configurations as follows:
```
Comment (appellj) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Rewrite this paragraph as: "During a WebSocket upgrade, the initial message is an HTTP request. To intercept this request and make the handshake explicitly with custom headers, the user must create an HTTP resource with WebSocket specific configurations as follows:"

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 5]
```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
...
+        webSocketUpgrade: {
+            upgradePath: "/{name}",
+            upgradeService: chatApp
+        }
+    }
+    upgrader(endpoint caller, http:Request req, string name) {
+    }
+}
+```
+Here `upgradeService` is a `WebSocketService`
```
Comment (appellj) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Insert "." at end of sentence

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 6]
```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
...
+
+**onOpen resource**: As soon as the WebSocket handshake is completed and the connection is established the `onOpen` resource is dispatched. This resource is only available in the `WebSocketService`.
```
Comment (appellj) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Insert "," after "established"

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md (hunk 4 above)
Comment (appellj) [reply stdlib/ballerina-http/src/main/ballerina/http/Package.md]: Please verify that I'm understanding the meaning correctly. It's the custom headers that intercept the message and make the handshake?

File: stdlib/ballerina-http/src/main/ballerina/http/Package.md [hunk 7]
```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
 
 See [Caching Example](https://ballerinalang.org/docs/by-example/caching), [HTTP Disable Chunking Example](https://ballerinalang.org/docs/by-example/http-disable-chunking).
 
+###WebSockets
+The package also provides support for WebSockets. There are two types of WebSocket endpoint: `WebSocketClient` and `WebSocketListener`. Both endpoints support all WebSocket frames. The `WebSocketClient` has a callback service.
```
Comment (appellj) [inline stdlib/ballerina-http/src/main/ballerina/http/Package.md]: change "endpoint" to "endpoints"

**Pull request merged**